### ISSUE TITLE
Think how to speed up syncing

### ISSUE THREAD

**Issue by chshersh** [kowainik/issue-wanted#134]
By @rashadg1030 calculations it takes ~6 hours to sync only all repos. And I can't even imagine how much time it will take to sync all issues as well... For testing purposes, we can have a configurable function that allows us to specify caching in a more flexible way. But we need some long-term strategy as well...

**Comment by rashadg1030** [comment]
@chshersh Maybe we should only go back to like 2015 or something. That will save time. I think right now we sync repos going all the way back to 2011 or so. I doubt users will want to go that far back, but on the other hand, there are old repos that might be popular.

**Comment by chshersh** [comment]
A repository can be created a long time ago, but it's still can be maintained. So we need all repositories eventually. We need 3-stage syncing for the proper implementation:

1. First initial fat sync that fetches **all** information. It's supposed to be launched only once when the application is started. But it would be good to implement it as a separate task if we want to perform total sync once per year.
2. A daily job that will sync for the last 1-2-3 (can be configurable) years of data.
3. When `Admin` is implemented, we should be able to force sync specific repo.

**Comment by rashadg1030** [comment]
@chshersh Oooh that's interesting. I'd like to know how to implement an Admin dashboard-type thing.
